src/arrayMethodJoin.js

```diff
@@ -4,8 +4,39 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let copy = '';
+
+    if (separator === null) {
+      for (let k = 0; k < this.length; k++) {
+        copy += 'null' + this[k];
+      }
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        copy += separator;
+      } else {
+        copy += separator + this[i];
+      }
+    }
+
+    let result = copy;
+
+    if (separator !== '') {
+      result = copy.slice(1);
+    }
+
+    if (typeof (separator) === 'object') {
+      result = copy.slice(15);
+    }
+
+    if (separator === null) {
+      result = copy.slice(4, -4);
+    }
+
+    return result;
```

Comment:
you have a very confusing algorithm, all you need is two checks:
1. check if the element is not equal to null and undefined, add it to the resulting string
2. after that, check if the elementr is not the last one - add a separator to the resulting string

that's all)